### PR TITLE
Add Top ~100 Polygon Tokens

### PR DESCRIPTION
**Pull request by 0xDaedalus**
First half of https://github.com/tallycash/extension/issues/1547

This list was derived by filtering the [official polygon tokenlist](https://github.com/maticnetwork/polygon-token-list/blob/master/src/tokens/allTokens.json) via the token addresses returned in [this query](https://dune.com/queries/909790?d=7)

For posterity - query pasted below:
```
SELECT SUM(usd_amount) as volume, token_b_address
from dex.trades
where block_time > '2021-05-01'
and block_time < '2021-06-14'
and project = 'Quickswap'
and usd_amount > 1
group by token_b_address
order by volume desc
LIMIT 200;
```

Only about 50% of the top 200 quickswap tokens by volume are on the polygon tokenlist - so we net out with ~100 new tokens.

### CODE REVIEW

**Review by greg-nagy — CHANGES_REQUESTED** [review]
This is a really good start, and I love that we are utilising dune for this ❤️

In the icon list I think we should do some clean up, before we merge this. Missing icons: btu, emon, ghst, iq (html instead of png), move.

Can't verify the compiled list of tokens, but I am comfortable trusting you with it, or defer the decesion to someone else. 

Non blocking questions: 
- Why 2021-05-01 - 2021-06-14?
- Does this also include cex trades? Or dex trades are representative enough?

File: file [hunk 1]
```diff
@@ -0,0 +1 @@
+Invalid token in file, can't find logo image - chains/137.json - [object Object][2K[1G[31merror[39m Command failed with exit code 65.
```
[the remark this answers is not in the document]
Comment (0xDaedalus) [reply file:1]: Trash - good catch

**Comment by 0xDaedalus** [comment]
> Why 2021-05-01 - 2021-06-14?

The more deterministic the list - the better - so someone can verify the list.

>  but I am comfortable trusting you with it

Someone should probably do a spot-check of at least a few - even if I'm a trusted party we should not necessarily trust the 3rd party source.  Its definitely possible that there are malicious tokens caught in that query (although highly unlikely).

> In the icon list I think we should do some clean up, before we merge this. Missing icons: btu, emon, ghst, iq (html instead of png), move.

Once again good catch - a great example of someone (me) blindly trusting 3rd party sources without verifying.

**Review by greg-nagy — CHANGES_REQUESTED** [review]
The not svg icons seem to be corrupted.

btu, emon, ghst, iq, move

github is not able to render them, and couldn't open them after download

We either should find icons manually, or remove them. My preference is on the former.

**Comment by Shadowfiend** [comment]
Approach feels sound on my end, though agree we should clean up any bad logos. If we need help sourcing the right ones perhaps we can reach out to the community?

**Comment by 0xDaedalus** [comment]
> Removing the icons from the list could work but I think it would be better to find the proper icons.

Why don't we add this as a community bounty - we have plenty of tokens in the app already that lack logos

**Comment by greg-nagy** [comment]
>> think it would be better to find the proper icons.

> Why don't we add this as a community bounty - we have plenty of tokens in the app already that lack logos

I think this is a pretty good idea. 

The thing I feel strongly about is not committing broken images if we know they are broken. 
I don't feel that strongly about finding the proper icons now, or just removing them and maybe creating a community bounty.

**Comment by 0xDaedalus** [comment]
Whoops! - sorry about that! 🙇 I thought I'd gotten all of them.

**Comment by greg-nagy** [comment]
@0xDaedalus do you want somebody to spot-check the token list or you feel good about merging?

**Comment by 0xDaedalus** [comment]
From a security POV yes we should probably not add tokens without spot-checking at least some of the set.

File: chains/137.json [hunk 2]
```diff
@@ -0,0 +1,564 @@
+[
+  {
+    "chainId": 137,
+    "name": "Ether - PoS",
+    "symbol": "ETH",
```
Comment (mhluongo) [inline chains/137.json:5]: Is there a reason we want to call this "ETH" and not "WETH"?

File: chains/137.json [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0x8f3cf7ad23cd3cadbd9735aff958023239c6a063",
+    "logoURI": "../images/dai.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Aave",
+    "symbol": "AAVE",
+    "decimals": 18,
+    "address": "0xd6df932a45c0f255f85145f286ea0b292b21c90b",
```
Comment (mhluongo) [inline chains/137.json:39]: PoS

File: chains/137.json [hunk 4]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0x53e0bca35ec356bd5dddfebbd1fc0fd03fabad39",
+    "logoURI": "../images/link.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Wrapped BTC",
+    "symbol": "WBTC",
+    "decimals": 8,
+    "address": "0x1bfd67037b42cf73acf2047067bd4f2c47d9bfd6",
```
Comment (mhluongo) [inline chains/137.json:55]: PoS

Comment (mhluongo) [reply chains/137.json:55]: Thinking about it, we could use a list extension here to label these as PoS if we want

File: chains/137.json [hunk 5]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0x831753dd7087cac61ab5644b308642cc1c33dc13",
+    "logoURI": "../images/quick.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Aavegotchi GHST Token",
+    "symbol": "GHST",
+    "decimals": 18,
+    "address": "0x385eeac5cb85a38a9a07a70c73e0a3271cfb54a7"
```
Comment (mhluongo) [inline chains/137.json:79]: PoS

File: chains/137.json [hunk 6]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0x8d1566569d5b695d44a9a234540f68D393cDC40D",
+    "logoURI": "../images/game.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Paint",
+    "symbol": "PAINT",
+    "decimals": 18,
+    "address": "0x7c28F627eA3aEc8B882b51eb1935f66e5b875714"
```
Comment (mhluongo) [inline chains/137.json:109]: FDV is $19k, we should remove

File: chains/137.json [hunk 7]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0xA3860f969075045D82de85B06bB665f93c4BAE32",
+    "logoURI": "../images/superbid.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "ClinTex",
+    "symbol": "CTI",
+    "decimals": 18,
+    "address": "0x8Ba941b64901E306667a287A370F145d98811096",
```
Comment (mhluongo) [inline chains/137.json:174]: Only 50 holders, I suggest we snip it

File: chains/137.json [hunk 8]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0x3e121107F6F22DA4911079845a470757aF4e1A1b",
+    "logoURI": "../images/fxs.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Frax",
+    "symbol": "FRAX",
+    "decimals": 18,
+    "address": "0x104592a158490a9228070E0A8e5343B499e125D0",
```
Comment (mhluongo) [inline chains/137.json:420]: PoS

File: chains/137.json [hunk 9]
```diff
@@ -0,0 +1,564 @@
...
+    "decimals": 18,
+    "address": "0xe86E8beb7340659DDDCE61727E500e3A5aD75a90",
+    "logoURI": "../images/zut.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Frax Share",
+    "symbol": "FXS",
+    "decimals": 18,
+    "address": "0x3e121107F6F22DA4911079845a470757aF4e1A1b",
```
Comment (mhluongo) [inline chains/137.json:412]: PoS

**Review by mhluongo — CHANGES_REQUESTED** [review]
My eyes are bleeding! but I spot-checked the list heavily.

I think we should be consistent about the "PoS" naming scheme, or move it to a schema extension that we label for future bridging metadata goodness. After that we should be gtg

File: chains/137.json [hunk 10]
```diff
@@ -0,0 +1,564 @@
...
+    "chainId": 137,
+    "name": "USD Coin",
+    "symbol": "USDC",
+    "decimals": 6,
+    "address": "0x2791bca1f2de4661ed88a30c99a7a9449aa84174",
+    "logoURI": "../images/usdc.svg"
+  },
+  {
+    "chainId": 137,
+    "name": "Tether USD",
```
[the remark this answers is not in the document]
Comment (0xDaedalus) [reply chains/137.json]: labled 👍

File: chains/137.json (hunk 2 above)
Comment (mhluongo) [reply chains/137.json:5]: Funny, the contract still say WETH but :shrug:

**Review by mhluongo — APPROVED** [review]
LFG :rocket:

**Pull request merged**